src/google/protobuf/compiler/parser.cc

```diff
@@ -931,6 +931,26 @@ bool Parser::ParseMessageBlock(DescriptorProto* message,
   if (message->reserved_range_size() > 0) {
     AdjustReservedRangesWithMaxEndNumber(message);
   }
+
+  // check if map_entry option was explicitly set
```

Comment:
rather than a comment to section off part of the message, please use a well named helper

Reply:
Done. I've added a `ValidateMessage` helper, analogous to existing `ValidateEnum`. The only check it performs is the check for an explicit `map_entry` option.

WDYT? Too generic?

Reply:
nah, looks about right.